File: src/main/java/uk/co/eelpieconsulting/feedlistener/IdBuilder.java
package uk.co.eelpieconsulting.feedlistener;

import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class IdBuilder {

    public String makeIdFor(String text) {
        return text.toLowerCase().trim().
                replaceAll("\\s", "-").
                replaceAll("[^\\-a-z0-9_]", "").
                replaceAll("--+", "-");
    }

    public String makeIdForChannel() {
        return UUID.randomUUID().toString();
    }

}

File: src/main/kotlin/uk/co/eelpieconsulting/feedlistener/IdBuilder.kt
package uk.co.eelpieconsulting.feedlistener

import org.springframework.stereotype.Component
import java.util.*

@Component
class IdBuilder {
    fun makeIdFor(text: String): String {
        val lowerCasedAndTrimmed = text.lowercase(Locale.getDefault()).trim { it <= ' ' }
        return lowerCasedAndTrimmed.replace("\\s".toRegex(), "-").replace("[^\\-a-z0-9_]".toRegex(), "")
            .replace("--+".toRegex(), "-")
    }

    fun makeIdForChannel(): String {
        return UUID.randomUUID().toString()
    }

}
